packages/auth/src/signature/middleware.ts

```diff
@@ -105,12 +106,20 @@ function validateSigInputComponents(
     if (component !== component.toLowerCase()) return false
   }
 
+  const isValidContentDigest =
+    ctx.request.body &&
+    Object.keys(ctx.request.body).length > 0 &&
+    !!ctx.headers['content-digest'] &&
+    (!sigInputComponents.includes('content-digest') ||
+      !verifyContentDigest(
+        JSON.stringify(ctx.request.body),
+        ctx.headers['content-digest'] as string
+      ))
+
   return !(
     !sigInputComponents.includes('@method') ||
     !sigInputComponents.includes('@target-uri') ||
-    (ctx.request.body &&
-      Object.keys(ctx.request.body).length > 0 &&
-      !sigInputComponents.includes('content-digest')) ||
+    isValidContentDigest ||
```

Comment:
```suggestion
    !isValidContentDigest ||
```
and maybe move it ahead of the `@method` check